db/compaction/subcompaction_state.h

```diff
@@ -240,12 +263,18 @@ class SubcompactionState {
   bool seen_key_ = false;
 
   // A flag determines if this subcompaction has been split by the cursor
-  bool is_split_ = false;
+  bool is_split_by_cursor_ = false;
 
   // We also maintain the output split key for each subcompaction to avoid
   // repetitive comparison in ShouldStopBefore()
   const InternalKey* local_output_split_key_ = nullptr;
 
+  // Grandparents' boundaries that fall into the subcompaction range, only
+  // populated for round-robin compaction priority
+  std::vector<InternalKey> grandparents_boundaries;
```

Comment:
Can callers use `compaction->grandparents()` instead of this? (If the optimization doesn't help, disregard this.)

Reply:
`compaction->grandparents()` is the grandparent for the whole input files, where `grandprants_boundaries` is just for this subcompaction range, belong to `compaction->grandparants()`